File: 1558-minimum-numbers-of-function-calls-to-make-target-array/1558-minimum-numbers-of-function-calls-to-make-target-array.cpp
class Solution {
public:
    int minOperations(vector<int>& nums) {
        int ans = 0, mx = 0;
        
        for(int x : nums) {
            ans += __builtin_popcount(x);
            mx = max(mx, x);
        }
        
        while(mx > 1) {
            ans++;
            mx /= 2;
        }
        
        return ans;
    }
};

File: 1558-minimum-numbers-of-function-calls-to-make-target-array/1558-minimum-numbers-of-function-calls-to-make-target-array.java
class Solution {
    public int minOperations(int[] nums) {
        int ans = 0;
        int mx = 0;

        for (int x : nums) {
            ans += Integer.bitCount(x);
            mx = Math.max(mx, x);
        }

        while (mx > 1) {
            ans++;
            mx /= 2;
        }

        return ans;
    }
}